plugins/cisco_asa/icon_cisco_asa/util/api.py

```diff
@@ -74,6 +74,36 @@ def create_address_object(self, name: str, object_type: str, address: str) -> di
             json_data={"name": name, "host": {"kind": object_type, "value": address}},
         )
 
+    def cli(self, commands: list) -> dict:
+        return self._call_api("POST", "cli", json_data={"commands": commands})
+
+    def get_blocked_hosts(self) -> list:
+        response = self.cli(["show shun"]).get("response", [])
+        blocked_hosts = []
+        if response and isinstance(response, list):
+            hosts = response[0].split("\n")
+            for host in hosts:
+                if host != "":
+                    split_host = host.split(" ")
+                    if (
+                        len(split_host) == 7
```

Comment:
Why is 7 important here; what is this supposed to check?

Reply:
The response for a single blocked host looks like this: `["shun (management) 1.1.1.1 2.2.2.2 444 555 6\n"]`
After splitting the response for this host, we should get 7 elements. This condition checks that we have obtained the correct number of elements and that those elements are not empty to make sure that the response is complete and contains all the necessary information.